lib/fluent/plugin/filter_record_transformer.rb

```diff
@@ -129,7 +124,9 @@ def reform(record, placeholder_values)
       placeholders = @placeholder_expander.prepare_placeholders(placeholder_values)
 
       new_record = @renew_record ? {} : record.dup
-      @keep_keys.each {|k| new_record[k] = record[k]} if @keep_keys and @renew_record
+      @keep_keys.each do |k|
+        new_record[k] = record[k] if record.has_key?(k)
+      end if @keep_keys and @renew_record
```

Comment:
Could you use `&&` instead of `and`?
This is not related with this patch but we want to unify condition style.